apps/data-pipeline/library-traffic-scraper/src/lib.ts

```diff
@@ -0,0 +1,118 @@
+import { database } from "@packages/db";
+import { libraryTraffic, libraryTrafficHistory } from "@packages/db/schema";
+import { load } from "cheerio";
+import fetch from "cross-fetch";
+
+type RawRespOK = {
+  message: "OK";
+  data: {
+    id: number;
+    name: string;
+    count: number;
+    percentage: number;
+    timestamp: string;
+  };
+};
+
+type RawRespErr = { error: string };
+
+async function fetchLocation(id: string): Promise<RawRespOK["data"] | null> {
+  const url = `https://www.lib.uci.edu/sites/all/scripts/occuspace.php?id=${id}`;
+  try {
+    const raw = await fetch(url).then((r) => r.text());
+
+    const parsedOnce = JSON.parse(raw);
+    const parsed = typeof parsedOnce === "string" ? JSON.parse(parsedOnce) : parsedOnce;
+
+    if (parsed.data && typeof parsed.data === "object") {
+      return (parsed as RawRespOK).data;
+    }
+
+    console.warn(`Location ID ${id} returned error: ${(parsed as RawRespErr).error}`);
+  } catch (err) {
+    console.error(`Unexpected error while fetching location ID ${id}:`, err);
+    throw err;
+  }
+  return null;
+}
+
+function extractLocationIds(pageHtml: string): string[] {
+  const $ = load(pageHtml);
+
+  let locationIds: string[] = [];
+
+  $("script").each((_, scriptTag) => {
```

Comment:
the use of `let` can be avoided; can you use `find` to get a matching `<script>` and then set `const locationIds` using that single found script? you can still throw if you didn't find that `<script>` that you wanted